presto-orc/src/main/java/com/facebook/presto/orc/OrcSelectiveRecordReader.java

```diff
@@ -721,7 +722,8 @@ public Page getNextPage()
                 blocks[i] = RunLengthEncodedBlock.create(columnTypes.get(columnIndex), constantValues[columnIndex] == NULL_MARKER ? null : constantValues[columnIndex], positionCount);
             }
             else if (!hasAnyFilter(columnIndex)) {
-                blocks[i] = new LazyBlock(positionCount, new OrcBlockLoader(getStreamReader(columnIndex), coercers[columnIndex], offset, positionsToRead, positionCount));
+                blocks[i] = new LazyBlock(positionCount, new OrcBlockLoader(getStreamReader(columnIndex), coercers[columnIndex],
+                        block -> updateMaxCombinedBytesPerRow(hiveColumnIndices[columnIndex], block), offset, positionsToRead, positionCount));
```

Comment:
LazyBlock can be loaded in a different thread and accessed much later. So could this cause threading issues, why or why not ?

Reply:
https://github.com/prestodb/presto/blob/master/presto-common/src/main/java/com/facebook/presto/common/Page.java#L294-L307, from what I see, these LazyBlocks seem to be loaded sequentially? I may not be the best person to answer this, maybe @mbasmanova knows better.

I just checked trino, seems they are doing something similar for LazyBlocks https://github.com/trinodb/trino/blob/fe608f2723842037ff620d612a706900e79c52c8/lib/trino-orc/src/main/java/io/trino/orc/OrcRecordReader.java#L424

Reply:
OrcBlockLoader will not work when there is multiple threads involved. So this fix looks ok to me.